### DISCUSSION TITLE
Is milvus not suitable for Kubernetes ?

### DISCUSSION THREAD

**Discussion by debashis1983** [milvus-io/milvus#23146] [Q&A and General discussion]
Hi,

We want to store 2.5 million vector of 768 dimensions in our organizations managed cloud. However every time when we tried to insert in milvus database using pymilvus lbrary,it is failing after inserting around 800k records. Upon communicating this issue to our devops colleagues, they told it is not suitable for k8s cluster because by design it will store hot data locally in the k8s node’s disk, instead of storing them in memory or mounted external volumes/disk.
Can you please confirm if this is the case. IF not please provide some pointers on the same.
Then we have to evaluate some other Vector DB for our use cases.

Regards,
Debashis

**Comment by debashis1983** [comment]
Or do we need to install it as a kubernetes operator?

**Reply by LoveEachDay** [reply]
Yes, milvus is design for cloud native and it will definity suitable for Kubernetes. How do you install a milvus instance in your Kubernetes cluster? What kind of storageclass do you have as a default storageclas?

**Reply by debashis1983** [reply]
![image](https://user-images.githubusercontent.com/4145637/229043519-2cdf7772-9a87-4cf4-8f56-0daca0374806.png)

**Comment by debashis1983** [comment]
![image](https://user-images.githubusercontent.com/4145637/229043791-4834d174-09b5-4394-812f-351a3429b122.png)

**Comment by debashis1983** [comment]
These is how we installed. Please let us know your recommendation.

Thanks.

**Reply by LoveEachDay** [reply]
What kind of index do you use? DiskAnn?

**Reply by debashis1983** [reply]
'index_type': "IVF_FLAT",

**Reply by LoveEachDay** [reply]
Could you provide the following command output:
 `kubectl get storageclass`

**Reply by debashis1983** [reply]
![image](https://user-images.githubusercontent.com/4145637/229072306-e35f1427-eaa2-4ec6-8015-ef30d417c018.png)

**Reply by debashis1983** [reply]
Another observation is the directory /var/lib/milvus/rdb_data is growing as more data insertion is going on.
Can you please tell me what it is storing.
![image](https://user-images.githubusercontent.com/4145637/229077369-6939a65b-efa1-41ee-baa0-9427acb64063.png)

**Reply by LoveEachDay** [reply]
Milvus will persist all the inserted into a message queue (rocksmq in the case of milvus standalone), then then milvus will consume the data into the object storage. 
So the data inside `/var/lib/milvus` is used by rocksmq, which is all the data you inserted.
